### PR TITLE
Added the clock target and required attributes

### PR DESCRIPTION
**Pull request by RameshIyyar**

### CODE REVIEW

**Comment by RameshIyyar** [comment]
Tested with Everest system MRW xml file. Clock target did not add into the Rainier system MRW XML file.

```
oscrefclk0 {
compatible = "ibm,power10-oscrefclk";
index = <0x00>;
ATTR_CLOCK_I2C_PATH = "/dev/i2c-0";
ATTR_FAPI_NAME = "NA";
ATTR_HWAS_STATE = [ 00 00 00 00 00 ];
ATTR_PHYS_BIN_PATH = [ 23 01 00 02 00 1A 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ];
ATTR_PHYS_DEV_PATH = "physical:sys-0/node-0/oscrefclk-0";
ATTR_POSITION = [ 00 01 ];
ATTR_TYPE = [ 1A ];
};

oscrefclk1 {
compatible = "ibm,power10-oscrefclk";
index = <0x01>;
ATTR_CLOCK_I2C_PATH = "/dev/i2c-1";
ATTR_FAPI_NAME = "NA";
ATTR_HWAS_STATE = [ 00 00 00 00 00 ];
ATTR_PHYS_BIN_PATH = [ 23 01 00 02 00 1A 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ];
ATTR_PHYS_DEV_PATH = "physical:sys-0/node-0/oscrefclk-1";
ATTR_POSITION = [ 00 00 ];
ATTR_TYPE = [ 1A ];
};
```

File: data/p10/filter_TargetsList.lsv [hunk 1]
```diff
@@ -31,3 +31,5 @@ unit-nx-power10
 unit-occ-power10
 # TPM target
 chip-NPCT750AAC
+# OSCREFCLK target
+chip-SI5332
```
Comment (ojayanth) [inline data/p10/filter_TargetsList.lsv:35]: Need @amitay input on pdbg side implementation.

File: attribute/attribute_target.c [hunk 2]
```diff
@@ -75,6 +75,7 @@ struct {
 
 	{ "TARGET_TYPE_TPM", "tpm", "tpm" },
 	{ "TARGET_TYPE_BMC", "bmc", "bmc" },
+	{ "TARGET_TYPE_OSCREFCLK", "oscrefclk", "oscrefclk" },
```
Comment (ojayanth) [inline attribute/attribute_target.c]: Commit message : Description looks contradictin , initiall mentioed fir guard purpose and the added additional attribute info. This target is not only use for guard purpose. Please update commit message accordingly

File: data/p10/pdbg_compatible_propMapping.lsv [hunk 3]
```diff
@@ -34,3 +34,5 @@ unit-occ-power10:ibm,power10-occ
 unit-thread:"ibm,power-thread", "ibm,power10-thread"
 # TPM target
 chip-NPCT750AAC:ibm,power10-tpm
+# OSCREFCLK target
```
Comment (ojayanth) [inline data/p10/pdbg_compatible_propMapping.lsv:37]: Need @amitay input on pdbg side implementation.

File: scripts/processMrw_bmc.pm [hunk 4]
```diff
@@ -50,6 +66,19 @@ sub buildBMCAffinity
             my $location_code = getLocationCode($targetObj, $target);
             $targetObj->setAttribute($target, "LOCATION_CODE", $location_code);
         }
+        elsif ($type eq "OSCREFCLK")
+        {
+            $oscrefclk++;
+            $osrefclk_instance_per_node++;
+            # Hardcoding the sys instance as "0" since we will have only one system
+            $targetObj->{targeting}{SYS}[0]{NODES}[$node]{OSCREFCLK}[$osrefclk_instance_per_node]{KEY}= $target;
```
Comment (ojayanth) [inline scripts/processMrw_bmc.pm:75]: we will have only one node?

Comment (RameshIyyar) [reply scripts/processMrw_bmc.pm:75]: we can have multi-node and that's why we are using the `$node` count in this path preparation. 

FYI, targets were read from MRW based on the order (same as FSP) so, I believe the order will be as expected in the multi-node system MRW but, if something is changed in the future then, we need to revisit based on the needs and currently we have only one node for p10 BMC system.

File: attribute/attribute_target.c (hunk 2 above)
Comment (RameshIyyar) [reply attribute/attribute_target.c]: Done

File: data/p10/filter_TargetsList.lsv (hunk 1 above)
Comment (RameshIyyar) [reply data/p10/filter_TargetsList.lsv:35]: Sorry, I am not getting it, what is the impact on the pdbg side with this name? or do you mean in the general?

The target type is picked based on the system MRW XML (Rainier and Everest).

File: data/p10/pdbg_compatible_propMapping.lsv (hunk 3 above)
Comment (RameshIyyar) [reply data/p10/pdbg_compatible_propMapping.lsv:37]: Yes, the pdbg target compatible property must be the same when adding the hardware unit on the pdbg side.

File: data/p10/filter_TargetsList.lsv (hunk 1 above)
Comment (ojayanth) [reply data/p10/filter_TargetsList.lsv:35]: You can ignore this comment , for got remove from here , pdbg_compatible_propMapping.lsv related changes. i place initiall her and forgot to remove.

File: scripts/processMrw_bmc.pm (hunk 4 above)
Comment (ojayanth) [reply scripts/processMrw_bmc.pm:75]: your commnet "we will have only one system" should c"hange to we will have only one node"?

Comment (RameshIyyar) [reply scripts/processMrw_bmc.pm:75]: sys and node are different right?
I think I meant "we will have only one sys target (system)" and the sys target can have more node target under that right?

**Comment by RameshIyyar** [comment]
Changed the OSCREFCLK target hierarchy as per pdbg expectation to perform the I2C read and write operation.

**Review by ojayanth — COMMENTED** [review]
Looking good to me, Need some clock firmware experts apparoval before merge.

File: attribute/old/attribute_target.c [hunk 5]
```diff
@@ -75,6 +75,7 @@ struct {
 
 	{ "TARGET_TYPE_TPM", "tpm", "tpm" },
 	{ "TARGET_TYPE_BMC", "bmc", "bmc" },
+	{ "TARGET_TYPE_OSCREFCLK", "oscrefclk", "oscrefclk" },
```
Comment (amitay) [inline attribute/old/attribute_target.c]: This is the old attribute code.  The new code uses libdtree.  So you will have to modify `libdtree/dtree_util.c`.

**Review by amitay — CHANGES_REQUESTED** [review]

File: attribute/old/attribute_target.c (hunk 5 above)
Comment (RameshIyyar) [reply attribute/old/attribute_target.c]: Thanks for pointing - I did not notice during rebase.

Added this line in `libdtree` and removed in the old code

**Review by ojayanth — APPROVED** [review]

**Review by rajez133 — APPROVED** [review]
Device tree generated is working for clock firmware

**Pull request merged**